internal/csi-addons/rbd/volumegroup.go

```diff
@@ -341,17 +341,40 @@ func (vs *VolumeGroupServer) ModifyVolumeGroupMembership(
 		}
 	}
 
-	// add the new volumes to the group
-	for _, id := range toAdd {
-		vol, getErr := mgr.GetVolumeByID(ctx, id)
-		if getErr != nil {
+	// resolve all volumes
+	volumes := make([]types.Volume, len(toAdd))
+	for i, id := range toAdd {
+		var vol types.Volume
+		vol, err = mgr.GetVolumeByID(ctx, id)
+		if err != nil {
 			return nil, status.Errorf(
 				codes.NotFound,
 				"failed to find a volume with CSI ID %q: %v",
 				id,
 				err)
 		}
 
+		//nolint:gocritic // need to call .Destroy() for all volumes
+		defer vol.Destroy(ctx)
```

Comment:
This deferred function is called when the outer function is exiting right ? 
not when the loop ends or all iterations are complete ? 
(I maybe mistaken)

Reply:
Yes, you are correct, this gets stacked up as it's called in for loop. removed it

Reply:
an other option, maybe cleaner?

```go
volumes := make([]types.Volume, len(toAdd))
defer func() {
    // need to call .Destroy() for all volumes
    for _, vol := range volume {
        vol.Destroy(ctx)
    }
}()

for i, id := range toAdd {
    ...
```

Reply:
i did think and reverted back, just before you reviewed the PR, will bring it back